### DISCUSSION TITLE
Using HTTPStatus instead of own defined constants

### DISCUSSION THREAD

**Discussion by sevdog** [encode/django-rest-framework#8167] [Ideas & Suggestions]
Starting from Python 3.5 the http module has defined an enum [`HTTPStatus`](https://docs.python.org/3/library/http.html#http.HTTPStatus) which has a collection of HTTP status codes just like that in [`status` module](https://github.com/encode/django-rest-framework/blob/f0a5b958a134e8cd94e3ef3263e8fa623ac9b82f/rest_framework/status.py#L30).

This enum is [already used](https://github.com/django/django/search?q=httpstatus) by Django itself.

There are some status codes which are present in this library module which are not already present into the standard lib enum, but they may be added at some point (and the missing ones are very few corner case status).

It could be a good idea to have this enum also here, thus avoiding the boilerplate constants definition?

**Comment by lovelydinosaur** [comment]
I'd probably need to consider the merits of this based on an actual pull request.
I think it's likely that it'd be a bit fiddly to do without breaking backwards compatibility, but... 🤷‍♂️

**Reply by sevdog** [reply]
I know that this may be a breaking change, it could be considered for a next major release.

I will try to make a PR about this when I have time.

**Reply by lovelydinosaur** [reply]
> I know that this may be a breaking change, it could be considered for a next major release.

I can't see us doing it if means a bunch of our users have to change their codebases. Just not worth the trade-off.

**Reply by sevdog** [reply]
It could be performed gradually, with warnings and documenting it.

A first step should be replacing it inside the library codebase, keeping the old module for backward compatibility with a deprecation warning. Maybe also moving the utility functions of status module into an other one (inside _utils_).

After a reasonable period than the module should be removed.

Since `http.HTTPStatus` is directly mantained by PSF and is part of Python and has more properties (like `.phrase` and `.description`) I belive thatit will add value to the lib.
